test/celeritas/em/UrbanMsc.test.cc

```diff
@@ -245,24 +245,19 @@ TEST_F(UrbanMscTest, msc_scattering)
                                        0.0465336,
                                        0.00708839};
 
-    static const real_type step[] = {0.00279169,
-                                     0.412343,
-                                     0.0376414,
-                                     0.078163296576415602,
-                                     0.031624394625545782,
-                                     0.002779271697902872,
-                                     0.00074215289000934838,
-                                     0.000031163160031423049};
-
-    constexpr unsigned int nsamples = std::end(step) - std::begin(step);
-    static_assert(nsamples == std::end(energy) - std::begin(energy),
-                  "Input sizes do not match");
+    constexpr unsigned int nsamples = std::end(energy) - std::begin(energy);
 
     // Mock SimStateData
     SimStateValue states_ref;
     auto          sim_state_data = make_builder(&states_ref.state);
     sim_state_data.reserve(nsamples);
 
+    // Calculate range instead of hardcoding to ensure step and range values
+    // are bit-for-bit identical when range limits the step. The first three
+    // steps are not limited by range
+    std::vector<double> step = {0.00279169, 0.412343, 0.0376414};
+    step.reserve(nsamples);
```

Comment:
This and the logic below is a little hard to follow: how about
```suggestion
    constexpr double step_is_range = -1;
    std::vector<double> step = {0.00279169, 0.412343, 0.0376414};
    step.resize(nsamples, step_is_range);
```
and then below, instead of  `if (i == step.size()) push_back(...)`, `if (step[i] == step_is_range) step[i] = ...`?

Better, or worse?

Reply:
better :)